src/core/mac/mac.cpp

```diff
@@ -817,8 +823,11 @@ void Mac::SentFrame(bool aAcked)
             ExitNow();
         }
 
-        sendFrame.GetDstAddr(destination);
-        mMle.RemoveNeighbor(destination);
+        break;
+
+    default:
+        assert(false);
```

Comment:
Why aren't other failures allowed? What if the platform ran out of memory or something else? I don't think this should be an assert.